modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_builder.py

```diff
@@ -377,11 +377,41 @@ def gen_reduce_expr(self):
                 self._dtype,
             )
 
+    class QuantileAggregate(CompoundAggregateWithColArg):
+        """
+        A QUANTILE aggregate generator.
+
+        Parameters
+        ----------
+        builder : CalciteBuilder
+            A builder to use for translation.
+        arg : List of BaseExpr
+            An aggregated values.
+        """
+
+        def __init__(self, builder, arg):
+            super().__init__(
+                "QUANTILE",
+                builder,
+                arg,
+                arg[0]._dtype
+                if is_datetime64_dtype(arg[0]._dtype)
+                else _get_dtype("float64"),
```

Comment:
It's pretty close to https://github.com/modin-project/modin/pull/6528/files#diff-532f2950a0930fecba54c8ab7a130cd9baa85d9320e8b46d9035931be52bc987R108.

Can this be generalized into a separate function?